lib/packages/docutils/rst.nim

```diff
@@ -1347,15 +1347,36 @@ proc match(p: RstParser, start: int, expr: string): bool =
     inc i
   result = true
 
-proc fixupEmbeddedRef(n, a, b: PRstNode) =
+proc safeProtocol*(linkStr: var string): string =
+  # Returns link's protocol and, if it's not safe, clears `linkStr`
+  result = ""
+  if scanf(linkStr, "$w:", result):
+    # if it has a protocol at all, ensure that it's not 'javascript:' or worse:
+    if cmpIgnoreCase(result, "http") == 0 or
+        cmpIgnoreCase(result, "https") == 0 or
+        cmpIgnoreCase(result, "ftp") == 0:
+      discard "it's fine"
```

Comment:
What determines whether a protocol is "trusted"? That is, why not "tel" (commonly used in telephone numbers), "ssh", etc.?

Reply:
This is related to URL-parsing. If you have tel/ssh URLs, you cannot write them as `[clickme](url)` anymore. That's perfectly ok. Alternatively only "javascript" links are really the harmful ones but I prefer the whitelist approach here.